src/main/java/org/truffleruby/core/module/ModuleFields.java

```diff
@@ -325,11 +325,16 @@ public boolean setConstantInternal(RubyContext context, Node currentNode, String
             final RubyConstant previous = constants.get(name);
             final boolean isPrivate = previous != null && previous.isPrivate();
             final boolean isDeprecated = previous != null && previous.isDeprecated();
-            final RubyConstant newValue = new RubyConstant(rubyModuleObject, value, isPrivate, autoload, isDeprecated);
+            final SourceSection sourceSection = currentNode != null ?  currentNode.getSourceSection() : null;
+            final RubyConstant newValue = new RubyConstant(rubyModuleObject, value, isPrivate, autoload, isDeprecated, sourceSection);
 
             if ((previous == null) ? (constants.putIfAbsent(name, newValue) == null) : constants.replace(name, previous, newValue)) {
                 newConstantsVersion();
-                return !autoload && previous != null;
+                if(!autoload){
```

Comment:
Missing spaces. Could you add a rule to catch them in `tool/lint.sh`?

Reply:
Fixed, I can add a lint for this but I think it'd be better to do a new PR for the lint.

Reply:
Sounds good.